#!/usr/bin/env python

PACKAGE='jsk_perception'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       name    type     level     description     default      min      max
gen.add("approximate_sync", bool_t, 0, "Enable Approximate Sync", False)
gen.add("queue_size", int_t, 0, "Queue size for message synchronization", 100, 1, 1000)
gen.add("slop", double_t, 0, "Slop value for approximate synchronizer", 0.1, 0.01, 10.0)

gen.add("face_padding", double_t, 0, "Padding factor for face rectangles", 0.0, -0.5, 0.5)
gen.add("face_threshold", double_t, 0, "Threshold for confidence of detected faces", 0.5, 0.0, 1.0)

exit(gen.generate(PACKAGE, "jsk_perception", "FacePoseEstimation"))
